### PR TITLE
Add expected total reward to RPC "getblocktemplate".

### PR DESCRIPTION
**Pull request by assylias**
Only works from V5 fork onward - returns 0 before that block.
It may make sense to do it only on testnet for now but I can't find a non-invasive solution.

### CODE REVIEW

File: src/rpc/core_rpc_server_commands_defs.h [hunk 1]
```diff
@@ -693,6 +693,7 @@ namespace cryptonote
       uint64_t height;
       uint64_t reserved_offset;
       std::string prev_hash;
+      uint64_t expected_reward;
```
Comment (hyc) [inline src/rpc/core_rpc_server_commands_defs.h]: keep the uint64_t's together. Move this up before prev_hash.

**Comment by assylias** [comment]
@hyc  updated - thanks.

File: src/cryptonote_core/tx_pool.cpp [hunk 2]
```diff
@@ -705,6 +705,7 @@ namespace cryptonote
       LOG_PRINT_L2("  added, new block size " << total_size << "/" << max_total_size << ", coinbase " << print_money(best_coinbase));
     }
 
+    expected_reward = best_coinbase;
```
Comment (moneromooo-monero) [inline src/cryptonote_core/tx_pool.cpp]: The actual coinbase will be coinbase, not best_coinbase, due to the break. best_coinbase is "best so far" to know when to break.

File: src/rpc/core_rpc_server_commands_defs.h [hunk 3]
```diff
@@ -692,6 +692,7 @@ namespace cryptonote
       uint64_t difficulty;
       uint64_t height;
       uint64_t reserved_offset;
+      uint64_t expected_reward;
```
Comment (moneromooo-monero) [inline src/rpc/core_rpc_server_commands_defs.h]: Bump minor RPC version when doing this.

**Comment by fluffypony** [comment]
@assylias just bumping this due to @moneromooo-monero's comments above

File: src/rpc/core_rpc_server_commands_defs.h (hunk 3 above)
Comment (assylias) [inline src/rpc/core_rpc_server_commands_defs.h]: @moneromooo-monero done, thanks.

**Comment by assylias** [comment]
@fluffypony  was away - should be fine now.

**Comment by NanoAkron** [comment]
@assylias Please re-push your response to @hyc's request, the repo doesn't seem to have updated

**Comment by assylias** [comment]
@NanoAkron  Argh, now hyc's comment is gone but moneromoo's comment is back...

**Pull request merged**